### PR TITLE
Issue #14937: Migrated section 2.3.3 Non-ASCII characters to follow chapter wise testing

### PR DESCRIPTION
**Pull request by Zopsss**
#14937

### CODE REVIEW

File: src/it/java/com/google/checkstyle/test/chapter2filebasic/rule233nonascii/NonASCIICharactersTest.java [hunk 1]
```diff
@@ -19,15 +19,12 @@
...
-
 import org.junit.jupiter.api.Test;
 
 import com.google.checkstyle.test.base.AbstractGoogleModuleTestSupport;
-import com.puppycrawl.tools.checkstyle.api.Configuration;
-import com.puppycrawl.tools.checkstyle.checks.AvoidEscapedUnicodeCharactersCheck;
 
-public class AvoidEscapedUnicodeCharactersTest extends AbstractGoogleModuleTestSupport {
+// -@cs[AbbreviationAsWordInName] Test should be named as its section name.
+public class NonASCIICharactersTest extends AbstractGoogleModuleTestSupport {
```
Comment (romani) [inline src/it/java/com/google/checkstyle/test/chapter2filebasic/rule233nonascii/NonASCIICharactersTest.java:27]: can we relax validation to do lowercase before compare or to do camel case(keep first capital, all other is lowercase) for words?

**Review by romani — CHANGES_REQUESTED** [review]
items

File: src/it/java/com/google/checkstyle/test/chapter2filebasic/rule233nonascii/NonASCIICharactersTest.java (hunk 1 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/chapter2filebasic/rule233nonascii/NonASCIICharactersTest.java]: done 
https://github.com/checkstyle/checkstyle/blob/5c7e58eae3a3610dd69b5985dfc25b5c4ece8b54/src/it/java/com/google/checkstyle/test/chapter2filebasic/rule233nonascii/NonAsciiCharactersTest.java#L26

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/chapter2filebasic/rule233nonascii/NonASCIICharactersTest.java]: also removed the suppression

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java [hunk 2]
```diff
@@ -2004,10 +2003,10 @@ private static String getExtractedRuleName(String ruleName) {
 
         for (String part : parts) {
             if (part.matches("[A-Za-z]+")) {
-                // Capitalize the first letter and make the rest words
+                // Capitalize the first letter and make the rest words lowercase
                 extractedRuleName
                         .append(part.substring(0, 1).toUpperCase(Locale.ENGLISH))
-                        .append(part.substring(1));
+                        .append(part.substring(1).toLowerCase(Locale.ENGLISH));
```
Comment (Zopsss) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java:2009]: had to modify this section based on @romani's suggestions here: https://github.com/checkstyle/checkstyle/pull/15094#discussion_r1649704947

**Review by romani — APPROVED** [review]
Thanks a lot, ok to merge

File: src/it/java/com/google/checkstyle/test/chapter2filebasic/rule233nonascii/NonAsciiCharactersTest.java [hunk 3]
```diff
@@ -36,20 +32,8 @@ protected String getPackageLocation() {
...
-            "36:47: " + getCheckMessage(AvoidEscapedUnicodeCharactersCheck.class, MSG_KEY),
-        };
-
-        final Configuration checkConfig = getModuleConfig("AvoidEscapedUnicodeCharacters");
-        final String filePath = getPath("InputAvoidEscapedUnicodeCharacters.java");
-
-        final Integer[] warnList = getLinesWithWarn(filePath);
-        verify(checkConfig, filePath, expected, warnList);
+        final String filePath = getPath("InputNonAsciiCharacters.java");
+        verifyWithConfigParser(new String[] {"AvoidEscapedUnicodeCharacters"}, filePath);
```
Comment (Zopsss) [inline src/it/java/com/google/checkstyle/test/chapter2filebasic/rule233nonascii/NonAsciiCharactersTest.java:36]: renamed the method

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**